### PR TITLE
Method field count

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/metadata/class-accessors.c [hunk 1]
```diff
@@ -138,3 +138,85 @@ mono_class_set_first_field_idx (MonoClass *klass, guint32 idx)
...
+
+void
+mono_class_set_method_count (MonoClass *klass, guint32 count)
+{
+	switch (klass->class_kind) {
+	case MONO_CLASS_DEF:
+	case MONO_CLASS_GTD:
+		((MonoClassDef*)klass)->method_count = count;
+		break;
+	case MONO_CLASS_GINST:
```
Comment (kumpera) [inline mono/metadata/class-accessors.c]: We probably should assert.

**Comment by kumpera** [comment]
When I tried to wrap this, it broke SRE big time

**Comment by vargaz** [comment]
The SRE tests in corlib run with this path.

**Comment by vargaz** [comment]
Add an assert.

**Pull request merged**